server/src/domain/asset/asset.service.ts

```diff
@@ -333,6 +333,9 @@ export class AssetService {
     await this.updateMetadata({ id, description, dateTimeOriginal, latitude, longitude, orientation });
 
     const asset = await this.assetRepository.save({ id, ...rest });
+    if (orientation) {
```

Comment:
Job scheduling can be complicated. Personally, I would see if there is a way to implement this in the job service itself. Basically, in the `onDone`. What should happen is what if the sidecar write job writes the orientation tag _then_ it should automatically trigger rotation.